router/batchrouter/asyncdestinationmanager/common/common.go

```diff
@@ -0,0 +1,161 @@
+	DestinationID       string
+}
+
+type AsyncPoll struct {
+	Config   map[string]interface{} `json:"config"`
+	ImportId string                 `json:"importId"`
+	DestType string                 `json:"destType"`
+}
+
+type ErrorResponse struct {
+	Error string
+}
+
+type Connection struct {
+	Source      backendconfig.SourceT
+	Destination backendconfig.DestinationT
+}
+type BatchedJobs struct {
+	Jobs       []*jobsdb.JobT
+	Connection *Connection
+	TimeWindow time.Time
+}
+
+type AsyncJob struct {
+	Message  map[string]interface{} `json:"message"`
+	Metadata map[string]interface{} `json:"metadata"`
+}
+
+type AsyncUploadT struct {
+	Config   map[string]interface{} `json:"config"`
+	Input    []AsyncJob             `json:"input"`
+	DestType string                 `json:"destType"`
+}
+
+type UploadStruct struct {
+	ImportId string                 `json:"importId"`
+	PollUrl  string                 `json:"pollURL"`
+	Metadata map[string]interface{} `json:"metadata"`
+}
+
+type MetaDataT struct {
+	CSVHeaders string `json:"csvHeader"`
+}
+type ImportParameters struct {
+	ImportId string    `json:"importId"`
+	PollUrl  *string   `json:"pollURL"`
+	MetaData MetaDataT `json:"metadata"`
+}
+
+type AsyncDestinationStruct struct {
+	ImportingJobIDs []int64
+	FailedJobIDs    []int64
+	Exists          bool
+	Size            int
+	CreatedAt       time.Time
+	FileName        string
+	Count           int
+	CanUpload       bool
+	UploadMutex     sync.RWMutex
+	URL             string
+	RsourcesStats   rsources.StatsCollector
+}
+
+type AsyncFailedPayload struct {
+	Config   map[string]interface{}   `json:"config"`
+	Input    []map[string]interface{} `json:"input"`
+	DestType string                   `json:"destType"`
+	ImportId string                   `json:"importId"`
+	MetaData MetaDataT                `json:"metadata"`
+}
+
+type FetchUploadJobStatus struct {
+	FailedJobURLs      string
+	Parameters         stdjson.RawMessage
+	ImportingList      []*jobsdb.JobT
+	PollResultFileURLs string
+}
+
+type EventStatMeta struct {
+	FailedKeys    []int64
+	ErrFailed     error
+	WarningKeys   []int64
+	ErrWarning    error
+	SucceededKeys []int64
+	ErrSuccess    error
+	FailedReasons map[int64]string
+}
+
+type GetUploadStatsResponse struct {
+	Status   string        `json:"status"`
+	Metadata EventStatMeta `json:"metadata"`
+}
+
+func GetTransformedData(payload stdjson.RawMessage) string {
+	return gjson.Get(string(payload), "body.JSON").String()
+}
+
+func GetBatchRouterConfigInt64(key, destType string, defaultValue int64) int64 {
+	destOverrideFound := config.IsSet("BatchRouter." + destType + "." + key)
+	if destOverrideFound {
+		return config.GetInt64("BatchRouter."+destType+"."+key, defaultValue)
+	} else {
+		return config.GetInt64("BatchRouter."+key, defaultValue)
+	}
+}
+
+/*
+Generates array of strings for comma separated string
+Also removes "" elements from the array of strings if any.
+*/
+func GenerateArrayOfStrings(value string) []string {
```

Comment:
Although a simple function, we can write a test for this.

Reply:
This being called from Poll and GetUploadStatus. Those are covered with test cases, are talking about test case for this util function exclusively ?